source/extensions/outlier_detection_monitors/common/monitor_base_impl.h

```diff
@@ -54,39 +54,62 @@ class LocalOriginEventsBucket : public TypedErrorsBucket<Upstream::Outlier::Loca
   bool match(const ExtResult&) const override;
 };
 
+/*
+    Base config for all types of monitors.
+*/
+class ExtMonitorConfig {
+public:
+  ExtMonitorConfig(const std::string& name, uint32_t enforce) : name_(name), enforce_(enforce) {}
+  void processBucketsConfig(
+      const envoy::extensions::outlier_detection_monitors::common::v3::ErrorBuckets& config);
+  void addErrorBucket(ErrorsBucketPtr&& bucket) { buckets_.push_back(std::move(bucket)); }
+
+  uint32_t enforce() const { return enforce_; }
+  const std::string& name() const { return name_; }
+  const std::vector<ErrorsBucketPtr>& buckets() const { return buckets_; }
+
+private:
+  std::string name_;
+  uint32_t enforce_{100};
+  std::vector<ErrorsBucketPtr> buckets_;
+};
+
+using ExtMonitorConfigSharedPtr = std::shared_ptr<ExtMonitorConfig>;
+
 class ExtMonitorBase : public ExtMonitor {
 public:
-  ExtMonitorBase(const std::string& name, uint32_t enforce) : name_(name), enforce_(enforce) {}
+  ExtMonitorBase(ExtMonitorConfigSharedPtr config)
+      : config_(std::move(config)),
+        enforce_runtime_key_("outlier_detection.enforcing_extension." + config_->name()) {}
```

Comment:
This runtime key could also be part of config.

Reply:
Moved to config.